internal/controller/clientpool/clientpool.go

```diff
@@ -44,11 +54,25 @@ func (cp *ClientPool) GetSDKClient(key ClientPoolKey) (sdkclient.Client, bool) {
 	cp.mux.RLock()
 	defer cp.mux.RUnlock()
 
-	c, ok := cp.clients[key]
-	if ok {
-		return c, true
+	info, ok := cp.clients[key]
+	if !ok {
+		return nil, false
 	}
-	return nil, false
+
+	// Check if any certificate is expired
+	if info.PEMCert != nil {
+		expired, err := cp.isCertificateExpiredSoon(info.PEMCert, 5*time.Minute)
+		if err != nil {
+			cp.logger.Error("Error checking certificate expiration", "error", err)
+			return nil, false
+		}
+		if expired {
+			cp.logger.Info("Certificate is expired or is going to expire soon")
+			return nil, false
```

Comment:
I might have missed this, but how do we trigger reloading the cert (assuming the secret has already been updated)? Or is the goal only to improve diagnostics through logging?